src/Http/Http/src/ReadOnlyPipeStream.cs

```diff
@@ -165,6 +165,7 @@ private async ValueTask<int> ReadAsyncInternal(Memory<byte> buffer, Cancellation
                 var readableBuffer = result.Buffer;
                 var readableBufferLength = readableBuffer.Length;
 
+                // TODO make this throw if the result is canceled
```

Comment:
Bug filed for this?

Reply:
https://github.com/aspnet/AspNetCore/pull/7618